tests/integration/callbacks/test_layout_paths_with_callbacks.py

```diff
@@ -0,0 +1,238 @@
+        def callback(value):
+            call_counts[counterId].value += 1
+            return {
+                "data": [
+                    {
+                        "x": ["Call Counter"],
+                        "y": [call_counts[counterId].value],
+                        "type": "bar",
+                    }
+                ],
+                "layout": {"title": value},
+            }
+
+        return callback
+
+    def generate_label_callback(id_):
+        def update_label(value):
+            call_counts[id_].value += 1
+            return value
+
+        return update_label
+
+    for chapter in ["chapter1", "chapter2", "chapter3"]:
+        app.callback(
+            Output("{}-graph".format(chapter), "figure"),
+            [Input("{}-controls".format(chapter), "value")],
+        )(generate_graph_callback("{}-graph".format(chapter)))
+
+        app.callback(
+            Output("{}-label".format(chapter), "children"),
+            [Input("{}-controls".format(chapter), "value")],
+        )(generate_label_callback("{}-label".format(chapter)))
+
+    dash_duo.start_server(app)
+
+    def check_chapter(chapter):
+        for key in dash_duo.redux_state_paths:
+            assert dash_duo.find_elements(
+                "#{}".format(key)
+            ), "each element should exist in the dom"
+
+        value = (
+            chapters[chapter][0]["{}-controls".format(chapter)].value
+            if chapter == "chapter3"
+            else chapters[chapter]["{}-controls".format(chapter)].value
+        )
+        # check the actual values
+        dash_duo.wait_for_text_to_equal("#{}-label".format(chapter), value)
+        wait.until(
+            lambda: (
+                dash_duo.driver.execute_script(
+                    "return document."
+                    'getElementById("{}-graph").'.format(chapter)
+                    + "layout.title.text"
+                )
+                == value
+            ),
+            TIMEOUT,
+        )
+
+        rqs = dash_duo.redux_state_rqs
+        assert rqs, "request queue is not empty"
+        assert all((rq["status"] == 200 and not rq["rejected"] for rq in rqs))
+
+    def check_call_counts(chapters, count):
+        for chapter in chapters:
+            assert call_counts[chapter + "-graph"].value == count
+            assert call_counts[chapter + "-label"].value == count
+
+    wait.until(lambda: call_counts["body"].value == 1, TIMEOUT)
+    wait.until(lambda: call_counts["chapter1-graph"].value == 1, TIMEOUT)
+    wait.until(lambda: call_counts["chapter1-label"].value == 1, TIMEOUT)
+    check_call_counts(("chapter2", "chapter3"), 0)
+
+    assert dash_duo.redux_state_paths == EXPECTED_PATHS["chapter1"]
+    check_chapter("chapter1")
+    dash_duo.percy_snapshot(name="chapter-1")
+
+    dash_duo.find_elements('input[type="radio"]')[1].click()  # switch chapters
+
+    # sleep just to make sure that no calls happen after our check
+    time.sleep(0.2)
```

Comment:
couldn't we get rid of these sleeps too by changing the order? Move the `percy_snapshot` call at least after the `wait.until` calls? In fact usually it seems like we call percy as late as possible, which in this case would mean after the `check_chapter` call.